rasa/core/policies/intent_ted_policy.py

```diff
@@ -239,7 +243,14 @@ def __init__(
             **kwargs,
         )
 
-        self.label_thresholds = label_thresholds
+        self.label_tolerance_thresholds = label_tolerance_thresholds
+        self.label_thresholds = (
```

Comment:
I'd put this into a separate public function. Here we kind of assume that you have to re-instantiate the model for prediction. If you wanted to train the model and then use the same instance for inference, you'd have to set this `label_thresholds` yourself, using the private method. Not that we need this, but it'd be nicer to have a separate function I think.

Reply:
We are trying to set an instance variable of the class which needs to be initialized to something in the constructor. I think it would be an overkill if we first declare it with a default value and then set it in a public function called after.

Reply:
Do you have a workaround in mind?

Reply:
@dakshvar22 how do you envision modifying tolerance at prediction time? Would that happen at instantiation or would this be done after instantiation. If the latter then I agree with @JEM-Mosig this should be a public method.

Reply:
The policy is loaded from the `load` method (which calls its own constructor to return an object) with the config used during training. So, as of now tolerance cannot be modified at prediction time because the trained config cannot be changed. Once that's possible, the `load` method should really be called with the new config (I am speculating here as it depends on how we really make it possible) and hence the constructor will take that into account. Based on this do you still recommend we refactor the logic into a separate public method?

Reply:
In that case (constructor is called with a new config) I think it is not needed to have a separate public method.